#21361 'Bestsellers collection is using ratingLimit as sql select limit'

yuriichayka opened the pull request:
Adding possibility for change range of bestseller items to be displayed
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
Drop-down was added with predefined set of values for changing number of items to be displayed
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. magento/magento2#21361: Bestsellers collection is using ratingLimit as sql select limit

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. ...
2. ...

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

yuriichayka committed:
magento/magento2#21361: Bestsellers collection is using ratingLimit as sql select limit
First commit. Basic functionality was implemented.

yuriichayka committed:
magento/magento2#21361: Bestsellers collection is using ratingLimit as sql select limit
Dropdown with predefined set of limits was added

yuriichayka committed:
magento/magento2#21361: Bestsellers collection is using ratingLimit as sql select limit
Fixing coding style problems

Commit:
Marked existing block as deprecated

yuriichayka committed:
#21361: Bestsellers collection is using ratingLimit as sql select limit
Applying requested changes

swnsma commented:
HI @yuriichayka 
Please check failed static tests

yuriichayka committed:
#21361: Bestsellers collection is using ratingLimit as sql select limit
Fixing code style, moving applied changes to \Magento\Sales\Block\Adminhtml\Report\Filter\Form for providing backward compatibility

yuriichayka committed:
#21361: Bestsellers collection is using ratingLimit as sql select limit
Changing depracated method, making method public in Magento\Reports\Block\Adminhtml\Bestsellers\Filter\Form

novikor commented on app/code/Magento/Sales/Block/Adminhtml/Report/Filter/Form/Coupon.php:
```diff
@@ -3,17 +3,28 @@
+ * @since      100.0.2
  */
-class Coupon extends \Magento\Sales\Block\Adminhtml\Report\Filter\Form
+class Coupon extends SalesBlockForm
```
@yuriichayka, did you just refactor \Magento\Sales\Block\Adminhtml\Report\Filter\Form\Coupon or there is some another sense?

yuriichayka replied:
- I added new $itemSortLimit construct parameter to Magento\Sales\Block\Adminhtml\Report\Filter\Form. So its options are initialized as arguments in di.xml.
- Since \Magento\Sales\Block\Adminhtml\Report\Filter\Form\Coupon is extended from it, I've changed parent construct call by adding $itemSortLimit array:
`parent::__construct($context, $registry, $formFactory, $orderConfig, $data, $itemSortLimit);`
- And some refactoring was done as well.

novikor commented on app/code/Magento/Sales/Block/Adminhtml/Report/Filter/Form.php:
```diff
@@ -3,57 +3,79 @@
         $fieldset = $this->getForm()->getElement('base_fieldset');
 
-        if (is_object($fieldset) && $fieldset instanceof \Magento\Framework\Data\Form\Element\Fieldset) {
+        if (is_object($fieldset) && $fieldset instanceof Fieldset) {
```
`is_object` check is redundant in this case

yuriichayka replied:
done

novikor commented on app/code/Magento/Sales/Block/Adminhtml/Report/Filter/Form/Coupon.php:
```diff
@@ -25,43 +36,44 @@ class Coupon extends \Magento\Sales\Block\Adminhtml\Report\Filter\Form
         $fieldset = $this->getForm()->getElement('base_fieldset');
 
-        if (is_object($fieldset) && $fieldset instanceof \Magento\Framework\Data\Form\Element\Fieldset) {
+        if (is_object($fieldset) && $fieldset instanceof Fieldset) {
```
`is_object` condition is redundant in this case

yuriichayka replied:
done

novikor reviewed (changes requested):
Okay, looks fine for now except 3 minor points.
More likely I`ll approve the PR ignoring Semantic and Health checks failures

yuriichayka commented:
Hello @novikor. 
Please check my answers.
If it is OK, I'll deploy requested minor changes.

yuriichayka committed:
#21361: Bestsellers collection is using ratingLimit as sql select limit
Fixing redunand conditions and parent:constructor call in app/code/Magento/Sales/Block/Adminhtml/Report/Filter/Form/Coupon.php

magento-engcom-team commented:
Hi @novikor, thank you for the review.
[ENGCOM-5472](https://jira.corp.magento.com/browse/ENGCOM-5472) has been created to process this Pull Request
:eight_spoked_asterisk: @novikor, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

engcom-Delta commented:
:heavy_check_mark: QA passed

sidolov committed:
Merge branch '2.3-develop' into 21361

slavvka commented:
@magento run all tests

slavvka commented:
@yuriichayka Could you please resolve the conflicts

slavvka commented:
@yuriichayka Also please cover the change with automated tests

novikor commented:
Thank you for your contribution, we do appreciate that.
Unfortunately, the PR is closed due to inactivity.
Please feel free to reopen it if you wish.
